### PR TITLE
Multiple code refactorings

### PR DESCRIPTION
**Pull request by cobaltgit**
- Use a more efficient dictionary comprehension rather than a for-loop on a list of KV tuples
- Use proper boolean logic rather than equality (`==`) comparisons
- Add type hinting to functions
- Multiple context managers on one line for ZipFile

### CODE REVIEW

File: CLI tools/CurseForge to MultiMC-Git.py [hunk 1]
```diff
@@ -9,28 +9,28 @@
 
 # Functions
 
-def remove_dir(path, description):
+def remove_dir(path: str, description: str) -> None:
```
Comment (Madis0) [inline CLI tools/CurseForge to MultiMC-Git.py:12]: What does `-> None` mean here?

File: CLI tools/CurseForge to Packwiz-Modrinth.py [hunk 2]
```diff
@@ -30,12 +30,12 @@ def extract_file(from_zip, from_file, to_path, from_desc, to_desc):
             print("Skipped " + from_desc + " copying to " + to_desc + ", didn't exist")
 
 mod_files = os.listdir(mods_path)
-if refresh_only == False:
+if not refresh_only:
```
Comment (Madis0) [inline CLI tools/CurseForge to Packwiz-Modrinth.py:33]: Interesting, didn't know that syntax in Python. Is there a performance difference as well?

File: CLI tools/MultiMC to Git.py [hunk 3]
```diff
@@ -7,24 +7,20 @@
...
 
     with zipfile.ZipFile(archive_path) as archive:
-        for zipinfo in archive.infolist():
-            if zipinfo.filename != to_path:
-                files.append((zipinfo.filename, archive.read(zipinfo.filename)))
+        files = {zipinfo.filename: archive.read(zipinfo.filename) for zipinfo in archive.infolist() if zipinfo.filename != to_path}
 
     with zipfile.ZipFile(archive_path, "w") as archive:
-        for filename, content in files:
+        for filename, content in files.items():
```
Comment (Madis0) [inline CLI tools/MultiMC to Git.py:21]: Python doesn't optimize it automatically?

**Review by Madis0 — CHANGES_REQUESTED** [review]
Very interesting, thanks for this!

File: CLI tools/MultiMC to Git.py (hunk 3 above)
Comment (RozeFound) [reply CLI tools/MultiMC to Git.py:21]: It's not optimizing anything, just making things more inline and messy. When I wrote this, I was aiming for readability and fitting existing code style. But this... I dunno, looks nasty.

File: CLI tools/CurseForge to Packwiz-Modrinth.py (hunk 2 above)
Comment (RozeFound) [reply CLI tools/CurseForge to Packwiz-Modrinth.py:33]: Nope, there is no performance difference, it's just (!=) operator, works same as in any other languages

Comment (Madis0) [reply CLI tools/CurseForge to Packwiz-Modrinth.py:33]: Would `if !refresh_only` work as well? I'm not too familiar with Python syntax

File: CLI tools/CurseForge to MultiMC-Git.py (hunk 1 above)
Comment (RozeFound) [reply CLI tools/CurseForge to MultiMC-Git.py:12]: It's return type. Just a good practice to specify types (params, return) even if it returns nothing (None, in this case)

File: CLI tools/CurseForge to Packwiz-Modrinth.py (hunk 2 above)
Comment (RozeFound) [reply CLI tools/CurseForge to Packwiz-Modrinth.py:33]: No, that's why `not` is used here. It's python variant of (!)

File: CLI tools/MultiMC to Git.py [hunk 4]
```diff
@@ -7,26 +7,20 @@
...
-        for zipinfo in archive.infolist():
-            if zipinfo.filename != to_path:
-                files.append((zipinfo.filename, archive.read(zipinfo.filename)))
-
-    with zipfile.ZipFile(archive_path, "w") as archive:
-        for filename, content in files:
-            archive.writestr(filename, content)
-        archive.write(from_path, to_path)
+    with zipfile.ZipFile(archive_path, "r") as inp, zipfile.ZipFile(archive_path, "w") as out:
+        files = {zipinfo.filename: inp.read(zipinfo.filename) for zipinfo in inp.infolist() if zipinfo.filename != to_path}
```
Comment (RozeFound) [inline CLI tools/MultiMC to Git.py:20]: It's nasty. And even if it optimizes anything, it's not worth creating such an unreadable mess.

File: CLI tools/MultiMC to Git.py (hunk 3 above)
Comment (RozeFound) [reply CLI tools/MultiMC to Git.py:21]: > I thought it chose the type by the contents.

He changed the type of var `files` from list to dictionary, and you can't just iterate over dict. You can iterate over `.values()` of this dict, or over `'.items()` which is list KeyValue's

File: CLI tools/MultiMC to Git.py (hunk 4 above)
Comment (RozeFound) [reply CLI tools/MultiMC to Git.py:20]: I would rather suggest revert changes and shorten file reading
```python

if (fname := zipinfo.filename) != to_path:
    files.append((fname, archive.read(fname)))

Comment (cobaltgit) [reply CLI tools/MultiMC to Git.py]: Well I pitched a dict comprehension which is cleaner than using KV tuples

Comment (RozeFound) [reply CLI tools/MultiMC to Git.py]: one beeeg line with multiple statements is cleaner than line by line loops?

**Review by osfanbuff63 — COMMENTED** [review]
There doesn't appear to be any new content on this branch now, I assume this is a mistake?

**Comment by Madis0** [comment]
I did what now? Merged a commit I created?

**Pull request merged**